This adds the possibility to control if FileLock of SoftFileLock is used with an environment variable.

Useful in case the filesystem doesn't support filelocking, but can probably
lead to problems if you are not careful in how you use it.
https://github.com/huggingface/datasets/issues/6744

Author: Viktor Rehnberg vikren@chalmers.se (Chalmers University of Technology)

diff --git a/src/datasets/utils/_filelock.py b/src/datasets/utils/_filelock.py
index 19620e6e..58f41a02 100644
--- a/src/datasets/utils/_filelock.py
+++ b/src/datasets/utils/_filelock.py
@@ -18,11 +18,15 @@
 import os
 
 from filelock import FileLock as FileLock_
-from filelock import UnixFileLock
+from filelock import SoftFileLock, UnixFileLock
 from filelock import __version__ as _filelock_version
 from packaging import version
 
 
+if os.getenv('HF_USE_SOFTFILELOCK', 'false').lower() in ('true', '1'):
+    FileLock_ = SoftFileLock
+
+
 class FileLock(FileLock_):
     """
     A `filelock.FileLock` initializer that handles long paths.
